ml-fundamentals/ml-problem-formulation-envisioning.md

```diff
@@ -0,0 +1,115 @@
+* Thinking about the solution as a black box, detail the function that a solution to this problem should perform to fulfil the objective and verify that the relevant data is available to solve the problem.
+  * One way of approaching this is by thinking about how a subject matter expert could solve the problem manually and the data that would be required; if a human subject matter expert is unable to use the available data to solve the problem, this is indicative that additional information is required and/or more data needs to be collected.
+* Based on the available data, define specific hypothesis statements - which can be proved
+or disproved - to guide the exploration of the data science team. Where possible, each hypothesis statement should have a clearly defined success criteria (e.g., *with an accuracy of over 60%*), however, this is not always possible - especially for projects where no solution to the problem currently exists. In these cases, the measure of success could be based on a subject matter expert verifying that the results meet their expectations.
+* Document all the above information, to ensure alignment between stakeholders and establish a clear understanding of the problem to be solved. Try to ensure that as much relevant domain knowledge is captured as possible, and that the features present in available data - and the way that the data was collected - are clearly explained, such that they can be understood by a non-subject matter expert.
+
+Once an understanding of the problem domain has been established, it may be necessary to break down the overall problem into smaller, meaningful chunks of work to maintain team focus and ensure a realistic project scope within the given timeframe.
+
+## Envisioning Checklist
+
+During envisioning sessions, the following may prove useful for guiding the discussion.
+
+### Problem Framing
+
+1. Define the objective in business terms.
+2. How will the solution be used?
+3. What are the current solutions/workarounds (if any)?
+4. How should performance be measured?
+5. Is the performance measure aligned with the business objective?
+6. What would be the minimum performance needed to reach the business objective?
+7. Are there any known constraints around non-functional requirements that would have to be taken into account? (e.g., computation times)
+8. Frame this problem (supervised/unsupervised, online/offline, etc.)
+9. Is human expertise available?
+10. How would you solve the problem manually?
+11. What work has been done in this area so far?
+12. Are there any restrictions on the type of approaches which can be used? (e.g., does the solution need to be completely explainable?)
+13. List the assumptions you or others have made so far. Verify these assumptions if possible.
+14. Define some hypothesis statements to be explored.
+15. Highlight and discuss any responsible AI concerns if appropriate.
```

Comment:
Other things that could be added here:
- How will this solution be maintained and monitored? 
- Does this solution need to fit within an existing system?